[3D] notify user if the 3D scene export failed

ptitjano opened the pull request:
## Description

A message is now displayed to notify that the export failed.

ptitjano committed:
qgs3dsceneexporter: Do not try to save if mObjects is empty

ptitjano committed:
qgs3dsceneexporter: Return save operation success state

ptitjano committed:
qgs3dmapscene: : Return export operation success state

nyalldawson commented on src/3d/qgs3dmapscene.cpp:
```diff
@@ -1045,7 +1045,12 @@ void Qgs3DMapScene::exportScene( const Qgs3DMapExportSettings &exportSettings )
+  const bool sceneSaved = exporter.save( exportSettings.sceneName(), exportSettings.sceneFolderPath() );
+  if ( !sceneSaved )
+  {
+    QgsMessageOutput::showMessage( tr( "3D exporter warning" ), tr( "The scene can not be exported." ), QgsMessageOutput::MessageText );
```
Can we use a messagebar message here instead? (It'd be nice to show a "success" message too, just like we do eg here: https://github.com/qgis/QGIS/blob/master/src/app/layout/qgslayoutdesignerdialog.cpp#L2268 )

ptitjano replied:
Done.

ptitjano committed:
qgsmap3dexportwidget: Return export scene operation success state

ptitjano committed:
qgs3dmapcanvaswidget: Display a message to report export success state

With this change, the messagebar is used to show a message if the
export operation failed or succeeded.

ptitjano committed:
qgs3dmapcanvaswidget: Give a proper name to QgsMap3DExportWidget

nyalldawson commented on src/app/3d/qgs3dmapcanvaswidget.cpp:
```diff
@@ -565,10 +565,21 @@ void Qgs3DMapCanvasWidget::exportScene()
+    const QString exportFilePath = QDir( exportSettings.sceneFolderPath() ).filePath( exportSettings.sceneName() + QStringLiteral( ".obj" ) );
+    if ( success )
+    {
+      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Successfully exported scene to <a href=\"%1\">%2</a>" ).arg( exportFilePath, QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Success, 0 );
```
```suggestion
      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Successfully exported scene to <a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( exportFilePath ), QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Success, 0 );
```

nyalldawson commented on src/app/3d/qgs3dmapcanvaswidget.cpp:
```diff
@@ -565,10 +565,21 @@ void Qgs3DMapCanvasWidget::exportScene()
+    }
+    else
+    {
+      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Unable to export scene to <a href=\"%1\">%2</a>" ).arg( exportFilePath, QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Warning, 0 );
```
```suggestion
      mMessageBar->pushMessage( tr( "Export 3D scene" ), tr( "Unable to export scene to <a href=\"%1\">%2</a>" ).arg( QUrl::fromLocalFile( exportFilePath ), QDir::toNativeSeparators( exportFilePath ) ), Qgis::MessageLevel::Warning, 0 );
```

ptitjano commented:
Fixed the issue introduced from the applied suggestion. A `toString()` call was missing.
